### PR TITLE
Change documentation for "default" tag in map configuration

### PR DESCRIPTION
**Pull request by orcunc**
As the resolution of the SUP :
https://hazelcast.atlassian.net/browse/SUP-550?focusedCommentId=102526

it was decided to update the documentation to make it clear that the "default" tag in map configuration means that it is used as "fallback" when the map's name that is being created does not match any entry

### CODE REVIEW

File: docs/modules/data-structures/pages/map-config.adoc [hunk 1]
```diff
@@ -33,9 +34,22 @@ For details on map backups, refer to xref:backing-up-maps.adoc[].
 
 For details on in-memory format, refer to xref:setting-data-format.adoc[].
 
-== Modifying the Default Configuration
+== The Default (Fallback) Map Configuration
+When a map is created, if the map name matches an entry in hazelcast.xml/yaml file, the values in the matching entry are used to overwrite initial values
+specified in <<map-configuration-defaults,Map Configuration Defaults>> section.
```
Comment (JamesHazelcast) [inline docs/modules/data-structures/pages/map-config.adoc:39]:
```suggestion
When a map is created, if the map name matches an entry in the `hazelcast.xml`/`hazelcast.yaml` file, the values in the matching entry are used to overwrite the initial values
discussed in the <<map-configuration-defaults,Map Configuration Defaults>> section.
```

File: docs/modules/data-structures/pages/map-config.adoc [hunk 2]
```diff
@@ -33,9 +34,22 @@ For details on map backups, refer to xref:backing-up-maps.adoc[].
...
 
-You can create a default configuration for all maps for your environment by modifying the map configuration block named "default" in your `hazelcast.xml`/`hazelcast.yaml` file. In the following example, we set expiration timers for map entries. Map entries that are idle for an hour will be marked as eligible for removal if the cluster begins to run out of memory. Any map entry older than six hours will be marked as eligible for removal. 
+This path also includes wildcard matches, meaning entries with “*” character
+
+If the map name does not match any entry in the hazelcast.xml/yaml file then
+
+- if “default” map configuration exists, the values under this tag are used to overwrite initial values. Therefore, "default" serves as a fallback.
+
+- if a “default” map configuration does not exist, the map is created with initial values
```
Comment (JamesHazelcast) [inline docs/modules/data-structures/pages/map-config.adoc:47]:
```suggestion
When a map name does not match any entry in the `hazelcast.xml`/`hazelcast.yaml` file then:

- If the `default` map configuration exists, the values under this entry are used to overwrite initial values. Therefore, `default` serves as a fallback.

- If a `default` map configuration does not exist, the map is created with initial values as discussed in <<map-configuration-defaults,Map Configuration Defaults>>.
```

File: docs/modules/data-structures/pages/map-config.adoc [hunk 3]
```diff
@@ -33,9 +34,22 @@ For details on map backups, refer to xref:backing-up-maps.adoc[].
 
 For details on in-memory format, refer to xref:setting-data-format.adoc[].
 
-== Modifying the Default Configuration
+== The Default (Fallback) Map Configuration
+When a map is created, if the map name matches an entry in hazelcast.xml/yaml file, the values in the matching entry are used to overwrite initial values
+specified in <<map-configuration-defaults,Map Configuration Defaults>> section.
 
-You can create a default configuration for all maps for your environment by modifying the map configuration block named "default" in your `hazelcast.xml`/`hazelcast.yaml` file. In the following example, we set expiration timers for map entries. Map entries that are idle for an hour will be marked as eligible for removal if the cluster begins to run out of memory. Any map entry older than six hours will be marked as eligible for removal. 
+This path also includes wildcard matches, meaning entries with “*” character
```
Comment (JamesHazelcast) [inline docs/modules/data-structures/pages/map-config.adoc:41]:
```suggestion
Maps that do not have any configuration defined use the default configuration. If you want to set a configuration that is valid for all maps, you can name your configuration as `default`. A user-defined default configuration applies to every map that does not have a specific custom map configuration defined with the map’s name. You can also use wildcards to associate your configuration with multiple map names. See the [configuration documentation](https://docs.hazelcast.com/hazelcast/5.5/configuration/using-wildcards) for more information about wildcards.
```

This is actually a snippet taken from Cloud's docs, and I think it's pretty well worded: https://docs.hazelcast.com/cloud/map-configurations

**Review by JamesHazelcast — COMMENTED** [review]
Thanks for putting this together @orcunc - a few suggestions from me to expand on your documentation slightly, taking some inspiration from Cloud's docs.

**Review by JamesHazelcast — APPROVED** [review]

**Review by oliverhowell — APPROVED** [review]

**Pull request merged**